programs/token/src/instructions/initialize_mint_2.rs

```diff
@@ -0,0 +1,72 @@
+use core::mem::MaybeUninit;
+
+use pinocchio::{
+    account_info::AccountInfo, instruction::{AccountMeta, Instruction, Signer}, program::invoke_signed, pubkey::Pubkey, ProgramResult
+};
+
+/// Initialize a new mint.
+///
+/// ### Accounts:
+///   0. `[WRITABLE]` Mint account
+pub struct InitilizeMint2<'a> {
+    /// Mint Account.
+    pub mint: &'a AccountInfo,
+    /// Decimals.
+    pub decimals: u8,
+    /// Mint Authority.
+    pub mint_authority: Pubkey,
```

Comment:
Here as well, it would be better to use a reference to `Pubkey` instead to avoid an extra copy.